content/configuration/executor.md

```diff
@@ -0,0 +1,147 @@
+    kubectl:
+      # ... trimmed ...
+```
+
+This can be later used by the communication mediums:
+
+```yaml
+communications:
+  'default-group':
+    slack:
+      channels:
+        'default':
+          bindings:
+            executors: # The order is important for merging strategy.
+              - kubectl-global       # The executor group name
+              - kubectl-team-a-only  # The executor group name
+          # ... trimmed ...
+```
+
+Multiple executor bindings, are merged together, see the [**merging strategy**](#merging-strategy) section for more details. For all available executor configuration properties, see the [**syntax**](#syntax) section.
+
+## Syntax
+
+```yaml
+# Map of executors. The `executors` property name is an alias for a given configuration.
+# Key name is used as a binding reference.
+#
+# Format: executors.<alias>
+executors:
+  'kubectl-read-only':
+    # Kubectl executor configuration.
+    kubectl:
+      namespaces:
+        # List of allowed Kubernetes Namespaces for command execution. The "all" keyword, allows all the Namespaces.
+        include:
+          - all
+        # List of ignored Kubernetes Namespace. It can be used only when `include: ["all"]`.
+        # It can contain a wildcard (*) that expands to zero or more arbitrary characters
+        # Example: `ignore: ["x", "y", "secret-ns-*"]`
+        ignore: []
+      # If true, enables `kubectl` commands execution.
+      enabled: false
+      # List of allowed `kubectl` commands.
+      commands:
+        # Configures which `kubectl` methods are allowed.
+        verbs: ["api-resources", "api-versions", "cluster-info", "describe", "diff", "explain", "get", "logs", "top", "auth"]
+        # Configures which K8s resource are allowed.
+        resources: ["deployments", "pods", "namespaces", "daemonsets", "statefulsets", "storageclasses", "nodes", "configmaps"]
+      # Configures the default Namespace for executing BotKube `kubectl` commands. If not set, uses 'default'.
+      defaultNamespace: default
+      # If true, enables commands execution from configured channel only.
+      restrictAccess: false
+```
+
+The default configuration for Helm chart can be found in [values.yaml](https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml).
+
+## Merging strategy
+
+BotKube takes into account only bindings for a given execution Namespace. For example:
+
+1. `@BotKube get po/botkube -n botkube` - collect `kubectl` executor bindings that include `botkube` or `all` Namespaces.
+2. `@BotKube get po -A` - collect all `kubectl` executor bindings that include `all` Namespace.
+3. `@BotKube get po` - first, we resolve the execution Namespace. For that, we collect all enabled `kubectl` executor bindings and check the `defaultNamespace` property. If property is not define, we use the `default` Namespace. With resolved execution Namespace, we run the logic define in the first step.
```

Comment:
1. I think that should be a separate paragraph or subparagraph - as it doesn't relate to general merging strategy, but rather `kubectl` executor and how it behaves. WDYT?
2. Probably bullet points would be better as this is not an ordered list 🙂

Reply:
1. I would leave it as it is for now. If we have more executors, we will have a clear use-case how to separate that 👍  For example, dedicated sections, files etc. We will also know what will be common for them.